lib/business/calendar.rb

```diff
@@ -167,11 +169,24 @@ def set_working_days(working_days)
           raise "Invalid day #{day}" unless DAY_NAMES.include?(normalised_day)
         end
       end
+      working_dates_names = @working_dates.map { |d| d.strftime("%a").downcase }
+      return if (working_dates_names & @working_days).none?
+      raise(ArgumentError, 'Working dates cannot be on working days')
```

Comment:
Our preferred code style for raising exceptions is like this:

```ruby
raise ArgumentError, 'Working dates cannot be on working days'
```

We'll look at adding a Rubocop rule to enforce this.